update-orphaning/index.html

```diff
@@ -1,206 +1,1086 @@
+    percentage = percentage.toFixed(1);
+    var oDetails;
+    if (aDetails1 && aDetails1[aIndex]) {
+      oDetails = aDetails1[aIndex];
+    } else if (aDetails2 && aDetails2[aIndex]) {
+      oDetails = aDetails2[aIndex];
+    }
+    var detailsText = "<b>" + percentage + "% (" + total + ") " + aIndex;
+    var linkText = "";
+    if (oDetails) {
+      detailsText = detailsText + " " + oDetails.code + "</b>: " + oDetails.desc;
+      linkText = getSearchLink(oDetails.code, aSrchOpt);
+    } else {
+      detailsText = detailsText + " Unknown</b>: Unknown";
+    }
+    document.getElementById(aPrefixID + "Link").innerHTML = linkText;
+    document.getElementById(aPrefixID + "Details").innerHTML = detailsText;
+  }
+
+  function getSearchLink(aSearchString, aSrchOpt) {
+    if (aSrchOpt === null || !aSearchString || aSearchString == "STATE PRESENT BUT INVALID" ||
+        aSearchString == "INVALID ERROR CODE" || aSearchString == "NO PING") {
+      return "";
+    }
+    var text = "Search dxr.mozilla.org for "
+    var searchString = 'regexp:"( |\\(|\\.|,)' + aSearchString + '( |;|\\)|,)"';
+    var path = "+path%3Atoolkit%2F";
+    var ext = "";
+    if (aSrchOpt) {
+      path = path + "mozapps%2Fupdate%2FnsUpdateService.js"; // Only nsUpdateService.js
+    } else {
+      ext = "+ext%3Acpp";
+    }
+    return "<a target='_blank' href='https://dxr.mozilla.org/mozilla-esr45/search?q=" +
+           searchString + path + ext + "&redirect=false'>" + text + aSearchString + "</a>";
+  }
 
   function populateDashboard(event) {
     var dataFile = null;
     if (event && event.type == "load") {
-      dataFile = lastWedDataFile;
+      dataFile = reportFile;
     } else {
-      dataFile = document.getElementById("weekly-dropdown").value;
+      dataFile = document.getElementById("weeklyDropdown").value;
     }
 
-    // Reset divs in case we can't find the data file.
-    {
-      document.getElementById("out-of-date-of-concern-summary").innerHTML = "<strong>No data found!</strong><br>(Failing URL: " + dataUrl + branch + dataFile + ")";
-      document.getElementById("out-of-date-updates-disabled-summary").innerHTML = "";
-      document.getElementById("up-to-date-summary").innerHTML = "";
-      document.getElementById("distribution-chart").innerHTML = "";
-      document.getElementById("update-ping-codes-chart").innerHTML = "";
-      document.getElementById("update-ping-code-description").innerHTML = "Hover over a bar below to see a description of the ping code.";
-      document.getElementById("version-distribution-chart-title").style.visibility = "visible";
-      document.getElementById("version-distribution-chart").innerHTML = "";
-      document.getElementById("version-distribution-chart").style.visibility = "visible";
-    }
+    clearContent();
 
+    function getJSONValue(aData, aName, aKey, aDefault) {
+      var result = aDefault;
+      try {
+        result = aData[aName][aKey];
+      } catch (e) {
+      }
```

Comment:
Added. Note: this is so it is possible to add new data without breaking reports using old data